backend/LeetWars.Core/LeetWars.Core.BLL/Interfaces/IUserService.cs

```diff
@@ -21,6 +21,8 @@ public interface IUserService
 
     public Task<UserFullDto> GetFullUserAsync(long id);
 
+    public Task<UserFullDto> GetUserByNickname(string nickname);

```

Comment:
it would be better to load UserId to leaderboard and use existed methods for loading user by Id